common/lib/src/realm_types.dart

```diff
@@ -52,14 +52,23 @@ enum RealmCollectionType {
   dictionary,
 }
 
+class RealmError extends Error {}
+
+class RealmUnsupportedError extends UnsupportedError implements RealmError {
+  RealmUnsupportedError(String message) : super(message);
+}
+
+class RealmUnsupportedSetError extends RealmUnsupportedError {
+  RealmUnsupportedSetError() : super('Cannot set late final field on realm object');
```

Comment:
It will be nice to have some suggestion in the text about how they can set this field if the setter is not supported. But this also would work for now.

Reply:
I would like to keep it simple. There are (typically) two kinds of late final fields: primary keys, and lists. It differs a bit what to do, ie. set it in ctor, or call `x.clear()..addAll([ ... ])`

Reply:
Yes it is better to be simple.